prow/gerrit/adapter/trigger.go

```diff
@@ -17,55 +17,57 @@ limitations under the License.
 package adapter
 
 import (
-	"strings"
 	"time"
 
+	"github.com/andygrunwald/go-gerrit"
 	"github.com/sirupsen/logrus"
-
 	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+
 	"k8s.io/test-infra/prow/config"
-	reporter "k8s.io/test-infra/prow/crier/reporters/gerrit"
-	"k8s.io/test-infra/prow/gerrit/client"
 	"k8s.io/test-infra/prow/pjutil"
 )
 
-// messageFilter builds a filter for jobs based on the messageBody matching the trigger regex of the jobs.
-func messageFilter(lastUpdate time.Time, change client.ChangeInfo, presubmits []config.Presubmit, latestReport *reporter.JobReport, logger *logrus.Entry) (pjutil.Filter, error) {
-	var filters []pjutil.Filter
-	currentRevision := change.Revisions[change.CurrentRevision].Number
+// presubmitContexts returns the set of failing and all job names contained in the reports.
+func presubmitContexts(failed sets.String, presubmits []config.Presubmit, logger *logrus.Entry) (sets.String, sets.String) {
+	allContexts := sets.String{}
+	for _, presubmit := range presubmits {
+		allContexts.Insert(presubmit.Name) // TODO(fejta): context, not name
+	}
+	failedContexts := allContexts.Intersection(failed)
```

Comment:
Just curious, is this for scenarios where a job was deleted after it failed? If yes, how can this be guarded from here to the point where it's re-kicked off?

Reply:
That is what this code does? The failedContexts set is the intersection of the set of jobs that still exist (allContexts set) and most recently reported as failed (failed set)

https://en.wikipedia.org/wiki/Intersection_(set_theory)

The point is to not rekick off deleted jobs. Retest will only retest failedContexts. By removing jobs that do not exist from the failedContexts set, we will not try retest them.

Reply:
ack